crates/kitsune_p2p/dht/src/spacetime/quantum.rs

```diff
@@ -0,0 +1,247 @@
+use super::*;
+
+/// Represents some particular quantum area of space. The actual DhtLocation that this
+/// coordinate corresponds to depends upon the space quantum size specified
+/// in the Topology
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    derive_more::Add,
+    derive_more::Sub,
+    derive_more::Display,
+    derive_more::From,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub struct SpaceQuantum(u32);
+
+impl SpaceQuantum {
+    /// The locations at either end of this quantum
+    pub fn to_loc_bounds(&self, topo: &Topology) -> (Loc, Loc) {
+        let (a, b): (u32, u32) = bounds(&topo.space, 0, self.0.into(), 1);
+        (Loc::from(a), Loc::from(b))
+    }
+}
+
+/// Represents some particular quantum area of time . The actual Timestamp that this
+/// coordinate corresponds to depends upon the time quantum size specified
+/// in the Topology
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    derive_more::Add,
+    derive_more::Sub,
+    derive_more::Display,
+    derive_more::From,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub struct TimeQuantum(u32);
+
+impl TimeQuantum {
+    /// The quantum which contains this timestamp
+    pub fn from_timestamp(topo: &Topology, timestamp: Timestamp) -> Self {
+        topo.time_quantum(timestamp)
+    }
+
+    /// The timestamps at either end of this quantum
```

Comment:
```suggestion
    /// The inclusive timestamps at either end of this quantum
```

... are these inclusive?